keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -63,6 +64,9 @@ val <T : Any> Resource<T>.name: ResourceName
 val <T : Any> Resource<T>.serviceAccount: String
   get() = metadata.getValue("serviceAccount").toString()
 
+val <T : Any> Resource<T>.application: String
+  get() = metadata.getValue("application").toString()
```

Comment:
Does this make it implement the `Applicationed` interface? Seems like it duck-types to it.

Reply:
The Applicationed/Monikered interfaces are being used on the way in to generate this value on metadata - but I think the fallback to `noapp` to ensure we have a value here isn't ideal

Reply:
removing the fallback and throwing an exception instead.